dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -678,12 +736,20 @@ ALWAYS_INLINE void Aggregator::executeImplBatch(
     /// Generic case.
 
     std::unique_ptr<AggregateDataPtr[]> places(new AggregateDataPtr[agg_size]);
+    size_t processed_rows = std::numeric_limits<size_t>::max();
```

Comment:
```suggestion
    std::optional<size_t> processed_rows;
```
It looks better to use `if (processed_rows) { *processed_rows - agg_process_info.start_row + 1, }`

Reply:
Good idea, refined.